src/models/GitHub.js

```diff
@@ -1,157 +1,158 @@
   getTasks(filter) {
     const extraParams = filter.updatedAt ? ` updated:>${filter.updatedAt}` : ' is:open'
-    const params = `?q=${encodeURIComponent(filter.query + extraParams)}`
-    const url = `${Config.githubApiUrl}/search/issues${params}&per_page=100`
-    return this.getTasksFromUrl(url)
+    const search = `${filter.query} ${extraParams}`
+    return this.getIssuesFromSearch(search)
   }
 
-  getTasksFromUrl(url, items = []) {
-    return this.get(url).then(({ json, headers }) => {
-      const allItems = items.concat(json.items)
-      const nextUrl = this.getNextUrl(headers)
-      if (nextUrl) {
-        return this.getTasksFromUrl(nextUrl, allItems)
+  getCurrentUser() {
+    const query = `{
+      viewer {
+        login
       }
+    }`
 
-      return { tasks: allItems.map(d => getTask(d)), nextUrl, currentUrl: url }
-    })
+    return this.graphql(query).then(result => result.viewer)
   }
 
-  // https://developer.github.com/v3/users/#get-the-authenticated-user
-  getCurrentUser() {
-    return this.get('user').then(result => result.json)
-  }
+  getIssuesFromSearch(search, endCursor, edges = []) {
+    return this.graphql(this.taskQuery(), { search, endCursor }).then(results => {
+      const allEdges = edges.concat(results.search.edges)
+
+      if (results.search.pageInfo.hasNextPage) {
+        const nextEndCursor = results.search.pageInfo.endCursor
+        return this.getIssuesFromSearch(search, nextEndCursor, allEdges)
+      }
 
-  // https://developer.github.com/v3/activity/notifications/#mark-a-thread-as-read
-  markAsRead(url) {
-    return this.patch(url, { ignoreBody: true })
+      return { tasks: allEdges.map(edge => this.transformEdgeToTask(edge)) }
+    })
   }
 
-  patch(relativeOrAbsoluteUrl, opts) {
-    let url = relativeOrAbsoluteUrl
-    if (url.indexOf('http') !== 0) {
-      url = `${Config.githubApiUrl}/${relativeOrAbsoluteUrl}`
+  graphql(query, variables = {}) {
+    const url = 'https://api.github.com/graphql'
+    const options = {
+      body: JSON.stringify({ query, variables }),
+      method: 'POST',
+      headers: this.getHeaders(),
     }
-    const options = opts || {}
-    options.headers = this.getHeaders()
-    return super.patch(url, options)
+
+    console.log(JSON.stringify(variables))
+
+    return fetch(url, options).then(response => {
+      if (!response.ok) {
+        return response.json().then(json => {
+          throw new Error(`${response.status}: ${JSON.stringify(json)}`)
+        })
+      }
+
+      return response.json().then(result => {
+        if (result.errors) {
+          throw new Error(`GraphQL Error: ${result.errors.map(e => e.message)}`)
+        }
+
+        return result.data
+      })
+    })
   }
 
   getHeaders() {
     if (!this.token) {
       this.token = GitHubAuth.getToken()
     }
     return {
-      Accept: 'application/vnd.github.v3+json',
-      Authorization: `token ${this.token}`,
+      Authorization: `bearer ${this.token}`,
```

Comment:
I saw they used bearer in the graphql docs, so I updated it to match that. I'm not sure if it matters though.